march_hardware_builder/src/hardware_builder.cpp

```diff
@@ -141,10 +144,14 @@ std::unique_ptr<march::IMotionCube> HardwareBuilder::createIMotionCube(const YAM
   std::ifstream imc_setup_data;
   imc_setup_data.open(ros::package::getPath("march_ems_projects").append("/sw_files/" + urdf_joint->name + ".sw"));
   std::string setup = convertSWFileToString(imc_setup_data);
-  return std::make_unique<march::IMotionCube>(
-      march::Slave(slave_index, pdo_interface, sdo_interface),
-      HardwareBuilder::createAbsoluteEncoder(absolute_encoder_config, urdf_joint),
-      HardwareBuilder::createIncrementalEncoder(incremental_encoder_config), setup, mode);
+
+  std::shared_ptr<march::IMotionCube> imc =
+      std::make_unique<march::IMotionCube>(march::Slave(slave_index, pdo_interface, sdo_interface),
```

Comment:
Why are you using `make_unique` if you want to create a `shared_ptr`?

Reply:
Oopsie :sweat_smile: